### PR TITLE
aiohttp exceptions can be normal responses

### PR DESCRIPTION
**Pull request by stj**
## What does this pull request do?

In `aiohttp` `HTTPExceptions` inherit from `Response` and can have a status
that is < 500. It even has a `HTTPOk` exception with a 200 status code
that can be raised.

Capturing these exceptions as errors in APM creates a fall's view of
response and service health.

This changes the `aiohttp` middleware to handle `HTTPExceptions` different
from other exceptions. It only captures `HTTPExceptions` equal or above a
500 status and handles all others like a success.

### CODE REVIEW

File: elasticapm/contrib/aiohttp/middleware.py [hunk 1]
```diff
@@ -78,21 +78,37 @@ async def handle_request(request, handler):
...
+            if exc.status_code < 500 and not should_trace:
+                raise
             if elasticapm_client:
-                elasticapm_client.capture_exception(
-                    context={"request": get_data_from_request(request, elasticapm_client.config, constants.ERROR)}
+                elasticapm.set_transaction_result("HTTP {}xx".format(exc.status_code // 100), override=False)
+                elasticapm.set_transaction_outcome(http_status_code=exc.status_code, override=False)
+                elasticapm.set_context(
+                    lambda: get_data_from_response(
+                        exc,
```
[the remark this answers is not in the document]
Comment (beniwohli) [reply elasticapm/contrib/aiohttp/middleware.py:93]: Looks like we run into https://github.com/PyCQA/pyflakes/issues/265 here. Guess we can suppress it with `# noqa F821`.

**Review by beniwohli — APPROVED** [review]
Great stuff, thanks for the PR!

**Review by basepi — APPROVED** [review]
Great work, thanks!

**Pull request merged**